File: DevEnvironment/Command/src/main/java/command/core/CommandParameters.java
package command.core;

import base.core.Conditions;
import command.interfaces.ICommandParameters;
import command.interfaces.IParameterSet;

/**
 * The CommandParameters class implements parameters of a command.
 */
public final class CommandParameters implements ICommandParameters {
    private final IParameterSet parameterSet;

    /**
     * The CommandParameters constructor.
     */
    public CommandParameters(IParameterSet parameterSet) {
        Conditions.validateNotNull(
            parameterSet,
            "The parameter-set of a command.");

        this.parameterSet = parameterSet;
    }

    /**
     * Gets the parameter-set of the command.
     */
    @Override
    public IParameterSet getParameterSet() {
        return this.parameterSet;
    }
}

File: SystemCore/Command/core/CommandParameters.cpp
#include "CommandParameters.h"
#include "CommandException.h"
#include "SmartPointers.h"

using namespace command;

/**
 * Creates a new command parameters.
 */
ICommandParametersSharedPtr CommandParameters::Make(IParameterSetSharedPtr parameterSet)
{
    return std::make_shared<CommandParameters>(parameterSet);
}

/**
 * The CommandParameters constructor.
 */
CommandParameters::CommandParameters(IParameterSetSharedPtr parameterSet)
{
    base::SmartPointers::Validate<IParameterSet>(parameterSet);
    m_parameterSet = parameterSet;
}

/**
 * The CommandParameters destructor.
 */
CommandParameters::~CommandParameters()
{
}

/**
 * Gets the parameter-set of the command.
 */
const IParameterSet& CommandParameters::GetParameterSet() const
{
    return *m_parameterSet;
}